makefile/Makefile

```diff
@@ -192,6 +192,8 @@ endif
 CXXFLAGS += $(CXXFLAGS_VECTORIZE)
 LDFLAGS += $(LDFLAGS_VECTORIZE)
 
+LDFLAGS +=  -Wl,--allow-multiple-definition
```

Comment:
please wrap this in some sort of flag, see, e.g., `UNIT_TESTS`, and do this only if "ALLOW_MULTIPLE_DEFINITIONS" is set to 1.